src/RibbonBase/RibbonBase.js

```diff
@@ -5,45 +5,57 @@ import Block from './Block';
+    render() {
+
+        return (
+            <div className="ontology-ribbon">
                 {
-                  group.data.map((slimitem) => {
-                    return (
-                      <Block
-                        {...slimitem}
-                        key={slimitem.id}
-                        onClick={() => this.props.onSlimSelect(slimitem.id)}
-                        isActive={slimitem.id === this.props.currentTermId}
-                      />
-                    );
-                  })
+                    this.props.groups.map((group) => {
+                        return (
+                            <div key={group.label} className='ontology-ribbon__strip'>
+                                {
+                                    group.data.map((slimitem) => {
+                                        return (
+                                            <Block
+                                                {...slimitem}
+                                                key={slimitem.id}
+                                                onClick={() => this.props.onSlimSelect(slimitem.id)}
+                                                onMouseEnter={() => this.props.handleSlimEnter(slimitem.id)}
+                                                onMouseLeave={() => this.props.handleSlimLeave(slimitem.id)}
+                                                isActive={slimitem.id === this.props.currentTermId}
+                                            />
+                                        );
+                                    })
+                                }
+                                <span className='ontology-ribbon__strip-label'
+                                      onClick={() => this.props.onDomainSelect(group.label)}
+                                      onMouseEnter={() => this.props.handleDomainEnter(group.label)}
+                                      onMouseLeave={() => this.props.handleDomainLeave(group.label)}
+                                >{group.label}</span>
+                            </div>
+                        )
+                    })
                 }
-                <span className='ontology-ribbon__strip-label'>{group.label}</span>
-              </div>
-            )
-          })
-        }
-      </div>
-    );
-  }
+            </div>
+        );
+    }
 }
 
 RibbonBase.defaultProps = {
-  groups: [],
+    groups: [],
 };
 
 RibbonBase.propTypes = {
-  currentTermId: PropTypes.string,
-  groups: PropTypes.arrayOf(PropTypes.shape({
-    data: PropTypes.arrayOf(PropTypes.shape({
-      id: PropTypes.string.isRequired,
+    currentTermId: PropTypes.string,
+    groups: PropTypes.arrayOf(PropTypes.shape({
+        data: PropTypes.arrayOf(PropTypes.shape({
+            id: PropTypes.string.isRequired,
+        })),
     })),
-  })),
-  onSlimSelect: PropTypes.func.isRequired
+    onSlimSelect: PropTypes.func.isRequired,
+    handleSlimEnter: PropTypes.func,
```

Comment:
Is it okay to rename the props `handle___` to `on___` to be a bit more consistent with other event listeners (onMouseEnter, onSlimSelect)? (Just for the RibbonBase module).

The props is really the API of a library. Probably more important for it follow convention than else where.

Reply:
on Renamed to onDomainEnter . . etc. 

I would typically keep as onMouseEnter / Leave, but RibbonBase has both.